modin/experimental/xgboost/test/test_xgboost.py

```diff
@@ -0,0 +1,172 @@
+# Apache License, Version 2.0 (the "License"); you may not use this file except in
+# compliance with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under
+# the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific language
+# governing permissions and limitations under the License.
+
+
+import pytest
+
+import modin.experimental.xgboost as xgb
+import modin.pandas as pd
+import numpy as np
+import xgboost
+
+from sklearn.datasets import (
+    load_iris,
+    load_diabetes,
+    load_digits,
+    load_wine,
+    load_breast_cancer,
+)
+from sklearn.metrics import accuracy_score, precision_score
+from numpy.random import uniform, randint
+
+import multiprocessing as mp
+
+
+num_cpus = mp.cpu_count()
+
+
+@pytest.mark.parametrize(
+    "num_round",
+    [2, 4, 8, 16, 32],
+    ids=["2", "4", "8", "16", "32"],
+)
+@pytest.mark.parametrize(
+    "num_actors",
+    [1, num_cpus, None],
+    ids=["1", "num_cpus", "None"],
+)
+@pytest.mark.parametrize(
+    "data",
+    [
+        (
+            "load_iris",
+            load_iris(),
+            {
+                "max_depth": 3,
+                "eta": 0.3,
+                "silent": 1,
+                "objective": "multi:softprob",
+                "num_class": 3,
+            },
+        ),
+        (
+            "load_diabetes",
+            load_diabetes(),
+            {
+                "colsample_bytree": uniform(0.7, 0.3),
+                "gamma": uniform(0, 0.5),
+                "learning_rate": uniform(0.03, 0.3),
+                "max_depth": randint(2, 6),
+                "n_estimators": randint(100, 150),
+                "subsample": uniform(0.6, 0.4),
+            },
+        ),
+        ("load_digits", load_digits(), {"max_depth": 6, "n_estimators": 200}),
+        (
+            "load_wine",
+            load_wine(),
+            {
+                "objective": "multi:softprob",
+                "eta": 0.4,
+                "max_depth": 3,
+                "gamma": 1.0,
+                "num_class": 3,
+            },
+        ),
+        (
+            "load_breast_cancer",
+            load_breast_cancer(),
+            {"objective": "binary:logistic", "eval_metric": ["logloss", "error"]},
+        ),
+    ],
+    ids=[
+        "load_iris",
+        "load_diabetes",
+        "load_digits",
+        "load_wine",
+        "load_breast_cancer",
+    ],
+)
+def test_dmatrix_train_predict_sklearn_datasets(data, num_actors, num_round):
+    name, dataset, param = data
+
+    X = dataset.data
+    y = dataset.target
+    dmatrix_xgb = xgboost.DMatrix(X, label=y)
+    with pytest.raises(AssertionError):
+        dmatrix_mxgb = xgb.DMatrix(X, label=y)
+
+    modin_X = pd.DataFrame(X)
+    modin_y = pd.DataFrame(y)
```

Comment:
Also, need to check that `pd.Series(y)` works correctly

Reply:
In this line currently use pd.Series(y)

Reply:
I mean, that we should test both of possible inputs for `modin_y`: `pd.DataFrame` and `pd.Series`

Reply:
Was added new parameter `type_modin_y` which have values [pd.DataFrame, pd.Series]